flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -240,7 +241,7 @@ public int exists(String pathInZooKeeper) throws Exception {
 		try {
 			return InstantiationUtil.deserializeObject(data, Thread.currentThread().getContextClassLoader());
 		} catch (IOException | ClassNotFoundException e) {
-			throw new Exception("Failed to deserialize state handle from ZooKeeper data from " +
+			throw new FlinkIOException("Failed to deserialize state handle from ZooKeeper data from " +
```

Comment:
The idea was to switch slowly to the new `FlinkExceptions`. However, what I just realize is that `FlinkIOException` does not inherit from `IOException` which is not good if you want to catch all `IOException`. Will revert it back to `IOException`.